test/src/unit-cppapi-profile.cc

```diff
@@ -0,0 +1,86 @@
+/**
+ * @file   unit-cppapi-profile.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2025 TileDB Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * Tests the C++ API for profile-related functionality.
+ */
+
+#include <test/support/src/helpers.h>
+#include <test/support/tdb_catch.h>
+
+#include "test/support/src/temporary_local_directory.h"
+#include "tiledb/sm/cpp_api/profile_experimental.h"
+#include "tiledb/sm/rest/rest_profile.h"
+
+using namespace tiledb;
+using namespace tiledb::test;
+
+struct ProfileCPPFx {
+  const std::string name_;
+  tiledb::sm::TemporaryLocalDirectory tempdir_;
+
+  ProfileCPPFx()
+      : name_(tiledb::sm::RestProfile::DEFAULT_NAME)
+      , tempdir_("unit_cppapi_profile") {
+  }
+};
+
+TEST_CASE_METHOD(
```

Comment:
I think these test cases should be augmented in some way to show the file path of the created profile